node/src/proxy_server/mod.rs

```diff
@@ -4934,6 +5879,97 @@ mod tests {
         );
     }
 
+    #[test]
+    fn new_http_request_creates_new_entry_inside_dns_retries_hashmap() {
```

Comment:
This sort of test is what I had in mind above when I commented on the zero-hop test.